src/snowflake/snowpark/_internal/open_telemetry.py

```diff
@@ -55,7 +55,11 @@ def open_telemetry_context_manager(func, dataframe):
             except Exception as e:
                 logger.warning(f"Error when acquiring span attributes. {e}")
             finally:
-                yield
+                try:
+                    yield
+                except Exception as e:
+                    cur_span.set_status(trace.Status(trace.StatusCode.ERROR, str(e)))
```

Comment:
Any test cases for this change?

Reply:
Sorry, I thought I added it in last PR, I have added the test